src/commands/mod.rs

```diff
@@ -1,25 +1,21 @@
 mod publish;
 
 use crate::commands::publish::Publish;
+use enum_dispatch::enum_dispatch;
+use anyhow::Result;
 use structopt::StructOpt;
 
 /// Available commands
+#[enum_dispatch(Run)]
 #[derive(StructOpt, Debug)]
 pub enum Command {
     /// Publishes this mod to BeatMods2
```

Comment:
Use of enum_dispatch seems to conflict with structopt, the description for the publish subcommand isn't displayed anymore. I'm not sure it's worth sacrificing subcommand descriptions for cleaner code.